src/main/java/EventScheduler.java

```diff
@@ -0,0 +1,120 @@
+package main.java;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EventScheduler {
+    private ArrayList<Event> eventList;
+    private TaskToEvent converter;
+    private TodoList todoList;
+    private GapFinder gapFinder;
+
+    public EventScheduler(TaskToEvent obj, GapFinder gapFinder){
+        this.eventList = new ArrayList<>();
+        this.converter = obj;
+        this.gapFinder = gapFinder;
+    }
+
+    public void addEvents(Event event){
+        this.eventList.add(event);
+    }
+
+    /**
+     *
+     * @param event event to be removed
+     * @return if the event has been removed
+     */
+    public boolean removeEvents(Event event){
+        if (this.eventList.contains(event)) {
+            this.eventList.remove(event);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Adding the tasks within the eventList to the todoList
+     */
+    public void addTaskToTodoList(){
+        for(Event evt: this.eventList){
+            if (!this.todoList.getUncompletedList().contains(evt.getTask()) &&
+                    !this.todoList.getCompletedList().contains(evt.getTask())) {
+                this.todoList.addTask(evt.getTask());
+            }
+        }
+    }
+
+    /**
+     * Converts a todoList of tasks into events and adds them to the eventList
+     * @param todoList  the todoList of tasks to be converted
+     */
+    public void toDoListToEvents(TodoList todoList){
+        for(Task task: todoList.getUncompletedList()) {
```

Comment:
currently only turns the uncompleted list into events, does this also need to convert the completed list?

Reply:
Oh yea ok I’ll fix it in a couple hours as I have class right now